src/ILCompiler/src/Program.cs

```diff
@@ -399,6 +401,21 @@ private int Run(string[] args)
             CompilationBuilder builder;
             if (_isWasmCodegen)
                 builder = new WebAssemblyCodegenCompilationBuilder(typeSystemContext, compilationGroup);
+            else if (_isReadyToRunCodeGen)
+            {
+                if (typeSystemContext.InputFilePaths.Count > 1)
+                {
+                    // Since ready-to-run currently adds to an existing IL image, we expect a single input
+                    throw new CommandLineException("Ready-to-run compilation only supports a single input assembly");
```

Comment:
Fine for now - longer term I've been thinking we could use a scheme where we would compile a set of assemblies in such a way that they would end up copied individually as the metadata and perhaps some tiny R2R header but that would forward to a final common dll (something like a R2R shared library) where the actual code / data would reside.

Reply:
Definitely. This is just a simple starting point.